File: src/2022/day-02/typescript/part-1.ts
// advent of code 2022 day 2 part 1

export function Part1(input: string) {
  const data = input.trim().split(/\n/).map((line) => line.split(/\s/));
  return data.map((p) => p[0]).map((p1, index) => {
    const p2 = data.map((p) => p[1])[index];
    if (p2 == 'Z') {
      if (p1 == 'A') return 0 + 3;
      else if (p1 == 'B') return 6 + 3;
      else return 3 + 3;
    } else if (p2 == 'Y') {
      if (p1 == 'A') return 6 + 2;
      else if (p1 == 'B') return 3 + 2;
      else return 0 + 2;
    } else {
      if (p1 == 'A') return 3 + 1;
      else if (p1 == 'B') return 0 + 1;
      else return 6 + 1;
    }
  }).reduce((a, b) => a + b);
}

File: src/2023/day-02/python/part_1.py
max_cubes = {"red": 12, "green": 13, "blue": 14}


def Part1(input):
    data = tuple(ln.strip() for ln in input.strip().split("\n"))
    s = 0
    for line in data:
        pos = True
        cubes = {"red": 0, "green": 0, "blue": 0}
        rest = line.split(":")
        gid = int(rest.pop(0)[5:])
        for g in [g.split(",") for g in rest[0].split(";")]:
            for v, k in [c.split() for c in g]:
                if max_cubes[k] < int(v):
                    pos = False
                cubes[k] = max(cubes[k], int(v))
        s += gid if pos else 0
    return s
